test/tools/cli/IonJavaCli.java

```diff
@@ -0,0 +1,1341 @@
+                        break;
+                    case STRUCT:
+                        ionStruct.add(event.getFieldName(),
+                                parseStruct(containerContext, compareContext, end, isFirst));
+                        break;
+                    case SEXP:
+                        ionStruct.add(event.getFieldName(),
+                                parseSequence(containerContext, compareContext, end, isFirst, ION_SYSTEM.newEmptySexp()));
+                        break;
+                }
+            } else if (eventType == EventType.CONTAINER_END) {
+                break;
+            } else if (eventType == EventType.STREAM_END) {
+                throw new IonException("Invalid struct: eventStream ends without CONTAINER_END");
+            } else {
+                IonValue cloneValue = event.getValue().clone();
+                cloneValue.removeFromContainer();
+                cloneValue.setTypeAnnotationSymbols(event.getAnnotations());
+                ionStruct.add(event.getFieldName(), cloneValue);
+            }
+        }
+        return ionStruct;
+    }
+
+    private static IonSequence parseSequence(ContainerContext containerContext,
+                                             CompareContext compareContext,
+                                             int end,
+                                             boolean isFirst,
+                                             IonSequence ionSequence) {
+        while (containerContext.increaseIndex() < end) {
+            Event event = isFirst ?
+                    compareContext.getEventStreamFirst().get(containerContext.getIndex()) :
+                    compareContext.getEventStreamSecond().get(containerContext.getIndex());
+            EventType eventType = event.getEventType();
+            if (eventType == EventType.CONTAINER_START) {
+                switch (event.getIonType()) {
+                    case LIST:
+                        ionSequence.add(parseSequence(containerContext, compareContext, end, isFirst, ION_SYSTEM.newEmptyList()));
+                        break;
+                    case SEXP:
+                        ionSequence.add(parseSequence(containerContext, compareContext, end, isFirst, ION_SYSTEM.newEmptySexp()));
+                        break;
+                    case STRUCT:
+                        ionSequence.add(parseStruct(containerContext, compareContext, end, isFirst));
+                        break;
+                }
+            } else if (eventType == EventType.CONTAINER_END) {
+                break;
+            } else if (eventType == EventType.STREAM_END) {
+                throw new IonException("Invalid ionSequence: eventStream ends without CONTAINER_END");
+            } else {
+                IonValue cloneValue = event.getValue().clone();
+                cloneValue.removeFromContainer();
```

Comment:
I believe cloning already removes the value from its container, so this line is probably redundant.